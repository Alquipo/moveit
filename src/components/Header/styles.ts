import styled, { css } from 'styled-components'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    height: 60px;
    background: ${theme.colors.backgroundBox};
    color: ${theme.colors.text};
    transition: background-color 0.3s, color 0.3s;

    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 0 30px;
  `}
`
