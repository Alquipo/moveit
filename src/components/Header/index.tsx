import { useContext } from 'react'
import { ThemeContext } from 'styled-components'
import Switch from 'react-switch'

import * as S from './styles'

export type HeaderProps = {
  toggleTheme(): void
}

const Header = ({ toggleTheme }: HeaderProps) => {
  const { colors, name } = useContext(ThemeContext)

  return (
    <S.Wrapper>
      <img src="img/logo-full.svg" alt="" />

      <Switch
        checked={name === 'Light'}
        onChange={toggleTheme}
        checkedIcon={false}
        uncheckedIcon={false}
        height={10}
        width={40}
        handleDiameter={15}
        onHandleColor={colors.grayLine}
        offHandleColor={colors.grayLine}
        offColor={colors.background}
        onColor={colors.background}
      />
    </S.Wrapper>
  )
}

export default Header
