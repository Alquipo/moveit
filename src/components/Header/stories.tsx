import { Story, Meta } from '@storybook/react/types-6-0'
import Header from '.'

export default {
  title: 'Header',
  component: Header
} as Meta

export const Default: Story = (args) => <Header {...args} />
