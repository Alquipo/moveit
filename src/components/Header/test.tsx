import { render, screen } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('should render the heading', () => {
    const { container } = render(<Header />)

    expect(screen.getByRole('heading', { name: /Header/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
