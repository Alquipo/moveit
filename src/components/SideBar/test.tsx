import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Header from '.'

jest.mock('react-switch', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return <img data-testid="Mock Switch">{children}</img>
  }
}))

describe('<Header />', () => {
  it('should render the Header correctly', () => {
    const { container } = renderWithTheme(<Header toggleTheme={jest.fn} />)

    expect(screen.queryByRole('button', { name: /Logo/i })).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /Logo/i })).toHaveAttribute(
      'src',
      `img/logo-full.svg`
    )

    expect(screen.getByTestId('Mock Switch')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
