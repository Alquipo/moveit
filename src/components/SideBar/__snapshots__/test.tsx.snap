// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header correctly 1`] = `
.c0 {
  height: 60px;
  background: #202024;
  color: #EEEEEE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 30px;
  -webkit-transition: background-color 0.3s ease-in-out,color 0.3s ease-in-out;
  transition: background-color 0.3s ease-in-out,color 0.3s ease-in-out;
}

<header
  class="c0"
>
  <img
    alt="Logo"
    role="button"
    src="img/logo-full.svg"
  />
  <img
    data-testid="Mock Switch"
  />
</header>
`;
